src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -295,10 +300,22 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 		}
 
 		if (messages.length !== 0) {
+			const previousMessages = messages.filter(message => {
+				return this.initialMessage
+					? timestampToNumber(message.timestamp) < timestampToNumber(this.initialMessage.timestamp)
+					: false;
+			});
+			if (previousMessages.length !== 0) this.messages = [...this.messages, ...previousMessages];
 			this.startIndex -= messages.length;
 
-			let newMessagesList = [...messages, ...this.messages];
-
+			let newMessagesList = [
+				...messages.filter(
+					message =>
+						!this.messages.includes(message) &&
+						message.messageId !== this.initialMessage?.messageId,
+				),
```

Comment:
includes will always return false since objects references are different

Reply:
about it, I've just checked without it and it is actually working as it must work